### ISSUE TITLE
Making your own "Small" LM

### ISSUE THREAD

**Issue by NirantK** [FuturePathAI/Learn-AI-Engineering#46]
Lecture 0: Making your own LM

First 1 hour:
1. Training your own GPT-2: Next word prediction
2. PyTorch: Perplexity, Loss Curves, Training loops
3. Introduce Weights and Biases

Use existing PyTorch.org tutorials as far as possible and cite them

Filename: Week4-Lecture0-PyTorch-Introduction.ipynb

Second hour: 
1. Fine-tuning a small LM notebook using Axolotl or Unsloth.ai 
- Use existing notebooks and tutorials
- Use 1-3B models or PEFT which fits the paid tier of Google Colab i.e. a maximum of A100
- Make sure the notebook runs "one-click" on Colab or other similar infra
